src/libqof/qof/gnc-datetime.hpp

```diff
@@ -153,10 +153,40 @@ class GncDateTime
     std::unique_ptr<GncDateTimeImpl> m_impl;
 };
 
+class GncDateFormat
+{
+public:
+    GncDateFormat (const char* fmt, const char* re) :
+    m_fmt(fmt), m_re(re) {}
+    const std::string m_fmt;
+private:
+    const std::string m_re;
+
+    friend class GncDateImpl;
+};
+
+/** Functor to check if a GncDateFormat refers to a specific date format.
+ * Intended to query GncDate::formats with std::find_if.
+ */
+struct GncDateFormatPred
+{
+    GncDateFormatPred( const std::string& name ) : m_name(name) {}
+    bool operator()( const GncDateFormat& v ) const
+    {
+        return (v.m_fmt == m_name);
+    }
+private:
+    const std::string m_name;
+};
+
 class GncDate
 {
-    public:/** Construct a GncDate representing the current day.
-        */
+    public:
+        /** Returns a vector with all the formats supported by the string constructor
```

Comment:
Doesn't *return* a vector, it *is* a vector.

Since it's a class member variable it should be c_formats.